lib/eslint.js

```diff
@@ -166,6 +168,14 @@ function addDeclaredGlobals(program, globalScope, config) {
         }
         variable.writeable = explicitGlobals[name];
     });
+
+    // mark all exported variables as such for no-unused-var
+    Object.keys(exportedGlobals).forEach(function(name) {
+        var variable = getVariable(globalScope, name);
```

Comment:
- es6 module enforces strict mode which doesn't allow creating global variables with `var`
- node.js has its own module environment which also doesn't create globals

So I think this command option is only for old browser libraries basically. I think the basics are covered. I need to test for all of them though :)